server/mscalendar/settings_daily_summary.go

```diff
@@ -0,0 +1,259 @@
+	value, err := s.store.GetSetting(userID, s.id)
+	if err != nil {
+		return nil, err
+	}
+
+	_, ok := value.(*store.DailySummaryUserSettings)
+	if !ok {
+		return nil, errors.New("current value is not a Daily Summary Setting")
+	}
+
+	return value, nil
+}
+
+func (s *dailySummarySetting) GetID() string {
+	return s.id
+}
+
+func (s *dailySummarySetting) GetTitle() string {
+	return s.title
+}
+
+func (s *dailySummarySetting) GetDescription() string {
+	return s.description
+}
+
+func (s *dailySummarySetting) GetDependency() string {
+	return s.dependsOn
+}
+
+func (s *dailySummarySetting) GetSlackAttachments(userID, settingHandler string, disabled bool) (*model.SlackAttachment, error) {
+	title := fmt.Sprintf("Setting: %s", s.title)
+	currentValueMessage := "Disabled"
+
+	actions := []*model.PostAction{}
+	if !disabled {
+		dsumRaw, err := s.Get(userID)
+		if err != nil {
+			return nil, err
+		}
+		dsum := dsumRaw.(*store.DailySummaryUserSettings)
+
+		currentH := "8"
+		currentM := "00"
+		currentAPM := "AM"
+		fullTime := "8:00AM"
+		currentEnable := false
+		currentTextValue := "Not set."
+
+		if dsum != nil {
+			fullTime = dsum.PostTime
+			currentEnable = dsum.Enable
+			splitted := strings.Split(fullTime, ":")
+			currentH = splitted[0]
+			currentM = splitted[1][:2]
+			currentAPM = splitted[1][2:]
+			enableText := "Disabled"
+			if currentEnable {
+				enableText = "Enabled"
+			}
+			currentTextValue = fmt.Sprintf("%s (%s) (%s)", dsum.PostTime, dsum.Timezone, enableText)
+		}
+
+		timezone, err := s.getTimezone(userID)
+		if err != nil {
+			if dsum != nil {
+				timezone = dsum.Timezone
+			} else {
+				timezone = "UTC"
+			}
+		}
+		fullTime = fullTime + " " + timezone
+
+		currentValueMessage = fmt.Sprintf("Current value: %s", currentTextValue)
+
+		actionOptionsH := model.PostAction{
+			Name: "H:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeHOptions(currentM, currentAPM, timezone),
+			DefaultOption: fullTime,
+		}
+
+		actionOptionsM := model.PostAction{
+			Name: "M:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeMOptions(currentH, currentAPM, timezone),
+			DefaultOption: fullTime,
+		}
+
+		actionOptionsAPM := model.PostAction{
+			Name: "AM/PM:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeAPMOptions(currentH, currentM, timezone),
+			DefaultOption: fullTime,
+		}
+
+		actions = []*model.PostAction{&actionOptionsH, &actionOptionsM, &actionOptionsAPM}
+
+		buttonText := "Enable"
```

Comment:
What purpose did the `if currentTextValue == "Not set."` block serve that is no longer necessary? Just want to understand the code.

Reply:
The idea was not to show the Enable button before a Daily Summary was set, but I thought that it would make sense to allow the enable with the default value.